examples/server/server.cpp

```diff
@@ -1476,20 +1504,32 @@ struct server_context {
             }
             tasks.push_back(std::move(task));
         };
-
-        static constexpr const char * error_msg = "\"prompt\" must be a string, an array of token ids or an array of prompts";
+        static constexpr const char * error_msg = "\"prompt\" must be a string, an array of token ids, a mixed array of strings and token ids, or an array of prompts";
         if (!data.contains("prompt")) {
             throw std::runtime_error(error_msg);
         }
-
         json prompt = data.at("prompt");
 
-        // if the prompt is a singleton (i.e. a string or a list of tokens), we only need to create single task
-        if (prompt.is_string() || json_is_array_of_numbers(prompt)) {
+        auto is_valid_singleton_array = [](const json& arr) {
+            bool has_number = false;
+            for (const auto& elem : arr) {
+                if (elem.is_number()) {
+                    has_number = true;
+                } else if (!elem.is_string()) {
+                    return false;
+                }
+            }
+            return has_number;
+        };
+
+        bool is_singleton = prompt.is_string() || (prompt.is_array() && is_valid_singleton_array(prompt));
```

Comment:
Just realized I accidentally left this "is_singleton" variable in here when I had meant to take it out. I will remove it in my next commit.